snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/ecommerce/events/AddToCart.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2015-2023 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.ecommerce.events
+
+import com.snowplowanalytics.core.constants.Parameters
+import com.snowplowanalytics.core.constants.TrackerConstants
+import com.snowplowanalytics.core.ecommerce.EcommerceAction
+import com.snowplowanalytics.snowplow.ecommerce.entities.Product
+import com.snowplowanalytics.snowplow.event.AbstractSelfDescribing
+import com.snowplowanalytics.snowplow.payload.SelfDescribingJson
+
+/**
+ * Track a product or products being added to cart.
+ *
+ * @param products - List of product(s) that were added to the cart.
+ * @param totalValue - Total value of the cart after the product(s) were added.
+ * @param currency - Currency used for the cart (ISO 4217).
+ * @param cartId - Cart identifier.
+ */
+class AddToCart @JvmOverloads constructor(
+
+    /**
+     * List of product(s) that were added to the cart.
+     */
+    var products: List<Product>,
+
+    /**
+     * The total value of the cart after this interaction.
+     */
+    var totalValue: Number,
```

Comment:
Nit: I think it would be nicer to have a `Cart` class that encapsulates the `totalValue`, `currency` and `cartId` properties similar like the `Product` class. At least to me it would make it more explicit what the properties refer to in this event and also in the `RemoveFromCart` one.